subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossBuildPerformanceTestRunner.groovy

```diff
@@ -36,7 +36,7 @@ abstract class AbstractCrossBuildPerformanceTestRunner extends AbstractGradleBui
     protected void defaultSpec(BuildExperimentSpec.Builder builder) {
         super.defaultSpec(builder)
         if (builder instanceof GradleBuildExperimentSpec.GradleBuilder) {
-            builder.invocation.distribution(gradleDistribution)
+            ((GradleBuildExperimentSpec.GradleBuilder) builder).invocation.distribution(gradleDistribution)
```

Comment:
Why is this cast needed? Shouldn't Groovy treat the `if` clause as if `builder` was already cast?

Reply:
I switched to `@CompileStatic`. Kotlin can infer the type after the `instanceof` check, I think the Groovy compiler can't - at least compilation failed for me.

Reply:
I saw this feature with Groovy first, I think it should work with `@CompileStatic` as well. This works for me in the Groovy Console in IntelliJ:

```groovy
@groovy.transform.CompileStatic
class MyType {
    static void main() {
        Object lajos = "alma"
        if (lajos instanceof String) {
            println lajos.length()
        }
    }
}
```

Reply:
That said, I don't think this is important enough to require a fix. I'm still curious why it won't compile without the cast.

Reply:
It seems to work fine in IDEA, though the Groovy compiler can't compile it: https://e.grdev.net/s/adjqb3njdwwoq/console-log?task=:internalPerformanceTesting:compileGroovy#L8